src/cdomains/intDomain.ml

```diff
@@ -813,20 +879,20 @@ struct
     | _ -> lift2_inj Integers.mul x y
   let div  = lift2 Integers.div
   let rem  = lift2 Integers.rem
-  let lt = lift2 Integers.lt
-  let gt = lift2 Integers.gt
-  let le = lift2 Integers.le
-  let ge = lift2 Integers.ge
+  let lt x y = lift2 Integers.lt x y
+  let gt x y = lift2 Integers.gt x y
+  let le x y = lift2 Integers.le x y
+  let ge x y = lift2 Integers.ge x y
   let bitnot = lift1 Integers.bitnot
```

Comment:
Similar issues as for the `lift2_special`:
```
    int x = 1;
    if(top) {
        x =2;
    }
    // x -> Not {0}([0,8])

    x = ~x; 
    // x -> Not {-1}([0,8])
```

Reply:
The range needs to be adjusted after every arithmetic operation on `Exc`. This is done in `lift2_inj`, not needed in `lift2`, and still missing in `lift1`.

Reply:
Why is it not needed in `lift2`? It is the one you call for shifts for example?

https://github.com/goblint/analyzer/blob/1e9119542b806aef0e96a86ed733f37e95f4e512/src/cdomains/intDomain.ml#L780-L794

Reply:
Fixed. I meant there's no computation done on `Excluded` in `lift2`, so no need to adjust the range.